pkg/providers/vsphere/vsphere.go

```diff
@@ -105,6 +105,12 @@ type ProviderGovcClient interface {
 	AddTag(ctx context.Context, path, tag string) error
 	ListCategories(ctx context.Context) ([]string, error)
 	CreateCategoryForVM(ctx context.Context, name string) error
+	DevicesInfo(ctx context.Context, datacenter, templateName string) (interface{}, error)
+	ValidateSize(ctx context.Context, datacenter, templateName string) error
```

Comment:
This doesn't strike me as the right place to do validation.

I'd think the only thing the vSphere client is used for is retrieving data and formatting it in a model of some kind. That model is then validated by logic residing in the vSphere package.